packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts

```diff
@@ -221,12 +217,21 @@ export function useGetJson(
   )
 
   // Separate the standard fields and annotations.
-  useEffect(() => {
-    if (query.data) {
-      setEntityMetadata(getStandardEntityFields(query.data))
-      setAnnotations(removeStandardEntityFields(query.data))
-    }
-  }, [query.data])
+  const entityMetadata = useMemo(
+    () =>
+      query?.data == undefined
+        ? undefined
+        : getStandardEntityFields(query.data),
+    [query.data],
+  )
+
+  const annotations = useMemo(
+    () =>
+      query?.data == undefined
+        ? undefined
+        : removeStandardEntityFields(query.data),
+    [query.data],
+  )
```

Comment:
These were previously state variables, so there was one "frame" where our data was loaded, so the returned `isLoading` was false, but these state variables hadn't updated yet, so they were still undefined. We can fix this by deriving the values from the `useQuery` result using `useMemo`.